src/components/Analysis/Purposes/LoadedPurposesList.jsx

```diff
@@ -0,0 +1,69 @@
+import React, { useMemo } from 'react'
+import cx from 'classnames'
+
+import List from 'cozy-ui/transpiled/react/MuiCozyTheme/List'
+import { useI18n } from 'cozy-ui/transpiled/react/I18n'
+import useBreakpoints from 'cozy-ui/transpiled/react/hooks/useBreakpoints'
+
+import {
+  computeAggregatedTimeseries,
+  sortTimeseriesByCO2GroupedByPurpose,
+  computeCO2Timeseries
+} from 'src/lib/timeseries'
+import { formatCO2 } from 'src/lib/trips'
+import { makeChartProps } from 'src/components/Analysis/helpers'
+import AnalysisListItem from 'src/components/Analysis/AnalysisListItem'
+import PieChart from 'src/components/PieChart/PieChart'
+
+const LoadedPurposesList = ({ timeseries }) => {
+  const { t } = useI18n()
+  const { isMobile } = useBreakpoints()
+
+  const aggregatedTimeseries = useMemo(
+    () => computeAggregatedTimeseries(timeseries),
+    [timeseries]
+  )
+  const timeseriesSortedByPurposes = useMemo(
+    () => sortTimeseriesByCO2GroupedByPurpose(aggregatedTimeseries),
+    [aggregatedTimeseries]
+  )
+  const totalCO2 = useMemo(() => computeCO2Timeseries(aggregatedTimeseries), [
+    aggregatedTimeseries
+  ])
+  const { data, options } = useMemo(
+    () => makeChartProps(timeseriesSortedByPurposes, 'purposes', t),
+    [t, timeseriesSortedByPurposes]
+  )
+
+  return (
+    <>
+      <div
+        className={cx('u-flex', {
+          ['u-flex-justify-end u-mr-2']: !isMobile,
+          'u-flex-justify-center u-mt-1': isMobile
+        })}
+      >
+        <PieChart
+          data={data}
+          options={options}
+          total={formatCO2(totalCO2)}
+          label={t('analysis.emittedCO2')}
+        />
+      </div>
+      <List>
+        {Object.entries(timeseriesSortedByPurposes).map(
+          (timeseriesSortedByPurpose, index) => (
+            <AnalysisListItem
+              key={index}
+              type="purposes"
+              sortedTimeserie={timeseriesSortedByPurpose}
+              totalCO2={totalCO2}
+            />
+          )
+        )}
+      </List>
+    </>
+  )
+}
+
+export default LoadedPurposesList
```

Comment:
that complex component needs to be tested, documented 🙏

Reply:
quel test verrais-tu ? Toute les méthodes utilisés dedans sont déjà testé, donc le traitement de la donnée est vérifié. Fonctionnellement il n'y a aucune logique dans la composant (pas de if par exemple, pas de pagination etc.) il ne fait qu'afficher 2 sous composants 🤔

Reply:
```
import React from 'react'
import { render } from '@testing-library/react'
import LoadedPurposesList from './LoadedPurposesList'
import { useI18n } from 'cozy-ui/transpiled/react/I18n'
import {
  computeAggregatedTimeseries,
  sortTimeseriesByCO2GroupedByPurpose,
  computeCO2Timeseries
} from 'src/lib/timeseries'
import { makeChartProps } from 'src/components/Analysis/helpers'

jest.mock('src/components/Analysis/helpers')
jest.mock('src/lib/timeseries')
jest.mock('cozy-ui/transpiled/react/hooks/useBreakpoints', () => () => ({
  isMobile: false
}))
jest.mock('cozy-ui/transpiled/react/I18n')
jest.mock(
  'src/components/Analysis/AnalysisListItem',
  () => ({ type, sortedTimeserie, totalCO2 }) => (
    <div
      data-testid="AnalysisListItem"
      data-test-type={type}
      data-timeserie={sortedTimeserie}
      data-total={totalCO2}
    />
  )
)
jest.mock(
  'src/components/PieChart/PieChart',
  () => ({ data, options, total, label }) => (
    <div
      data-testid="PieChart"
      data-test={data}
      data-options={options}
      data-total={total}
      data-label={label}
    />
  )
)

describe('LoadedPurposesList', () => {
  const timeseries = ['timeseries']
  const aggregatedTimeseries = 'computeAggregatedTimeseries'
  const t = x => x
  const firstTimeserieSortedByPurposes = {
    timeseries: [],
    totalCO2: 22
  }
  const timeseriesSortedByPurposes = {
    firstTimeserieSortedByPurposes,
    WORK: {
      timeseries: [],
      totalCO2: 23
    }
  }

  beforeEach(() => {
    useI18n.mockReturnValue({ t })
    computeAggregatedTimeseries.mockReturnValue(aggregatedTimeseries)
    sortTimeseriesByCO2GroupedByPurpose.mockReturnValue(
      timeseriesSortedByPurposes
    )
    computeCO2Timeseries.mockReturnValue(48)
    makeChartProps.mockReturnValue({ data: 'data', options: 'options ' })
  })

  it('should contain a AnalysisListItem with correct timeseriesSortedByPurposes and total CO2', () => {
    // When
    const { getAllByTestId } = render(<LoadedPurposesList />)

    // Then
    expect(
      getAllByTestId('AnalysisListItem')[0].getAttribute('data-test-type')
    ).toEqual('purposes')
    expect(
      getAllByTestId('AnalysisListItem')[0].getAttribute('data-timeserie')
    ).toEqual('firstTimeserieSortedByPurposes,[object Object]')
    expect(
      getAllByTestId('AnalysisListItem')[0].getAttribute('data-total')
    ).toEqual('48')
  })

  it('should contain a PieChart with correct total CO2', () => {
    // When
    const { getByTestId } = render(<LoadedPurposesList />)

    // Then
    expect(getByTestId('PieChart').getAttribute('data-label')).toEqual(
      'analysis.emittedCO2'
    )
    expect(getByTestId('PieChart').getAttribute('data-options')).toEqual(
      'options'
    )
    expect(getByTestId('PieChart').getAttribute('data-test')).toEqual('data')
    expect(getByTestId('PieChart').getAttribute('data-total')).toEqual('48 kg')
  })

  it('should computeAggregatedTimeseries from timeseries', () => {
    // When
    render(<LoadedPurposesList timeseries={timeseries} />)

    // Then
    expect(computeAggregatedTimeseries).toHaveBeenCalledWith(timeseries)
  })

  it('should sortTimeseriesByCO2GroupedByPurpose from computeAggregatedTimeseries', () => {
    // When
    render(<LoadedPurposesList timeseries={timeseries} />)

    // Then
    expect(sortTimeseriesByCO2GroupedByPurpose).toHaveBeenCalledWith(
      aggregatedTimeseries
    )
  })

  it('should computeCO2Timeseries from computeAggregatedTimeseries', () => {
    // When
    render(<LoadedPurposesList timeseries={timeseries} />)

    // Then
    expect(computeCO2Timeseries).toHaveBeenCalledWith(aggregatedTimeseries)
  })

  it('should makeChartProps from computeAggregatedTimeseries', () => {
    // When
    render(<LoadedPurposesList timeseries={timeseries} />)

    // Then
    expect(makeChartProps).toHaveBeenCalledWith(
      'sortTimeseriesByCO2GroupedByPurpose',
      'purposes',
      t
    )
  })
})
```

Reply:
wow merci pour le test complet 👍 (les "it should" auraient suffit hein :) )

à discuter peut-être ensemble, et à fixer peut-être sur un paper mais : 
- Je comprends pas trop l'intérêt de mocker à tout va, on ne teste alors le comportement normal.
- Je ne comprends pas l'intérêt de ces tests ? Si je comprends bien on teste juste qu'un composant return un child spécifique avec une prop spécifique... ? 🤔 Etant donné qu'il n'y a aucune "logique" ni condition dans le composant, est-ce qu'on ne peut pas faire confiance à React pour ça ?
- Je ne comprends pas l'intérêt de tester qu'un composant appelle des méthodes spécifiques. Ce qu'on souhaite plutôt, éventuellement, c'est qu'il affiche un certain rendu non ? (le quoi plutôt que le comment ?)
- Pour les tests sur le PieChart je verrais plus un snapshot du composant éventuellement, comme sur cozy-ui

Bref à discuter de vive voix je pense, à mon avis on sort du cadre de la review...